unlock-app/src/hooks/useLockSettings.ts

```diff
@@ -0,0 +1,47 @@
+import { SubgraphService } from '@unlock-protocol/unlock-js'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+
+interface LockSettingsProps {
+  lockAddress: string
+  network: number
+}
+
+export function useLockSettings() {
+  const web3Service = useWeb3Service()
+  const subgraph = new SubgraphService()
+
+  const getIsRecurringPossible = async ({
+    lockAddress,
+    network,
+  }: LockSettingsProps) => {
+    const lock = await subgraph.lock(
+      {
+        where: {
+          address_in: [lockAddress],
+        },
+      },
```

Comment:
Why do you use the graph here???

Reply:
`getLock` from `web3service` is slower

Reply:
restore `const lock = await web3Service.getLock(lockAddress, network)` with parameters we need